src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,48 +1,84 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+interface Props {
+  onHideTodo: () => void;
+  todo: Todo;
+}
+
+export const TodoModal: React.FC<Props> = ({ onHideTodo, todo }) => {
+  const { id, title, completed } = todo;
+
+  const [user, setUser] = useState<User | null>(null);
+  const fetchedUser = async () => {
+    try {
+      if (todo.userId) {
+        const data = await getUser(todo.userId);
+
+        setUser(data);
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.log(error);
```

Comment:
would be nice to show some error to our user

Reply:
> would be nice to show some error to our user

thank you so much! All suggestions are appropriate!)